compose-builder/tui-generator.sh

```diff
@@ -0,0 +1,299 @@
+#!/usr/bin/env bash
+
+# /*******************************************************************************
+#  * Copyright 2022 Shantanoo Desai <shantanoo.desai@gmail.com>
+#  *
+#  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+#  * in compliance with the License. You may obtain a copy of the License at
+#  *
+#  * http://www.apache.org/licenses/LICENSE-2.0
+#  *
+#  * Unless required by applicable law or agreed to in writing, software distributed under the License
+#  * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  * or implied. See the License for the specific language governing permissions and limitations under
+#  * the License.
+#  *
+#  *******************************************************************************/
+
+# generator.sh - provide a Terminal User Interface to make generation of different EdgeX Foundry
+# services more comprehensible.
+# Author: Shantanoo (Shan) Desai <shantanoo.desai@gmail.com>
+
+# Map the available options in `make compose` either as standalone strings or array of strings
+
+# Individual Compose Descriptions that may not fall into EdgeX-Foundry Architecture Components
+# Or may be an initial "out-of-the-box" configuration
+
+# Check if whiptail exists to render the UI, if not, exit script
+WHIPTAIL=$(which whiptail)
+
+if [ -z $WHIPTAIL ]
+then
+    echo "This script requires whiptail to render the UI."
+    exit 1;
+fi
+
+MAKE=/usr/bin/make
+
+SELECTED_DEVSERVICES=()
+SELECTED_APPSERVICES=()
+SELECTED_BUS=()
+SELECTED_OTHERS=()
+
```

Comment:
One concern I have is that `tput` will mess up the terminal layout once the script exits the Whiptail TUI. This is because LINES and COLUMNS are generally available in a bash session and changing them might affect the spacing. I initially had some weird layout of my terminal after adding these variables but was able to reset it back using `tput clear`

Reply:
based on the current changes @ajcasagrande can you please check to see if `tput` is changing some layout and spacing in your terminal after the script is executed?